src/Styleguide/Elements/Box.tsx

```diff
@@ -34,33 +35,51 @@ import {
 } from "styled-system"
 import { Flex, FlexProps } from "./Flex"
 
-const hover = css`
-  &:hover {
-    border-color: ${color("black60")};
-  }
-`
-
 export interface BorderBoxProps
   extends FlexProps,
     SpaceProps,
     MaxWidthProps,
     WidthProps,
     HeightProps {
   hover?: boolean
+  responsive?: boolean
+  hasSiblings?: boolean
 }
 
 export const BorderBox = styled(Flex).attrs<BorderBoxProps>({})`
   border: 1px solid ${color("black10")};
   border-radius: 2px;
-  ${props => props.hover && hover};
+  padding: ${space(2)}px;
   ${styledSpace};
   ${maxWidth};
   ${width};
   ${height};
+  ${({ hover }) =>
+    hover &&
+    css`
+      :hover {
+        border-color: ${color("black60")};
+      }
+    `};
+  ${({ responsive }) =>
+    responsive &&
+    media.sm`
+      padding: ${space(3)}px;
+    `};
+  ${({ hasSiblings }) =>
+    hasSiblings &&
+    css`
+      :not(:first-child) {
+        border-top-left-radius: 0;
+        border-top-right-radius: 0;
+      }
+      :not(:last-child) {
+        border-bottom: 0;
+        border-bottom-left-radius: 0;
+        border-bottom-right-radius: 0;
+      }
+    `};
```

Comment:
Could you talk me through this? It seems like this is a specific implementation detail of a certain use case that's being added to a more generic component. 

The `hasSibilings` prop here is expecting that the borderbox appears in a stack with one item on top of another. That isn't necessarily the only usecase for this. I'm also uncertain of the responsibility of the `responsive` prop and why it would be used.

Reply:
I was also curious. This feels a bit like API creep around something that can naturally be handled without explicit values being passed.

Reply:
Yeah these are options to extend the behaviour of BorderBox. `hasSiblings` makes the borders seamless between sibling BorderBoxes, and `responsive` adds responsive padding.

I was taking the same approach as the `hover` prop, assuming that was the idiomatic way to add optional behaviour, but it sounds like you were both surprised to see these kinds of options being added. What would be the more idiomatic way to do this? Make a new component?

Reply:
Ah, so `hasSiblings` allows things to be stackable, so that the bottom border on one component doesn't double up with the top border on a box below, yah? Rather than adding a new API method It might be more flexible to piggy back on styled-system and make the border more generic; e.g., 

```jsx
<BorderBox borderBottom={0}>
```

Since `BorderBox` is a primitive, I feel like adding `responsive` as a configuration prop might be some unnecessary overhead in terms of what `BorderBox` is responsible for. Two possible approaches would be to configure padding from the outside using the `<Responsive>` component, e.g., 

```js
<Responsive>
  {({xs} => 
     <BorderBox p={xs ? 2 : 0}>...</BorderBox>
  }}
<Responsive>
```

Or worry about padding from within the content of the `BorderBox`, like 

```jsx
<BorderBox>
   <SomeLocalStyledComponentWithResponsivePadding />
</BorderBox>
```

Reply:
In general, I think we should design for responsiveness. A parent shouldn't need to worry if a child it's consuming is responsive or not, unless it's implicitly implied (i.e SmallComponent, LargeComponent). By default we should be able to consume something like `BorderBox` in our given context and assume it to work correctly. The fact that this component changes its padding on mobile makes me wonder if `borderbox` currently has invalid padding on mobile. I think these two should probably behave inherently the same.

Ultimately, if the component generally responds a certain way on different breakpoints that should be encoded in the component directly, not passed down on every usage of the component. The parent should only handle responsive behavior when it has special needs. 

Having a separate primitive for a group of connecting border-box components does feel right. I think once we sort out the difference in padding behavior between this component and border-box this component will be a nice edition. At that point I'd propose just renaming it to `StackableBorderBox`. 

Works as is for now though. @ds300 thanks for your patience through this! 😃

Reply:
Re `StackableBorderBox` -- one thought is that it could also exist as an export from `BorderBox` rather than an additional file.